lib/rules/var-spacing.rule.js

```diff
@@ -10,90 +10,59 @@
-}
+var validatePunctuators = require('../util/validatePunctuators');
+var findNextPunctuators = require('../util/findNextPunctuators');
 
-function isMultipleLineDeclaration(node, token) {
-  return node.loc.start.line < token.loc.start.line;
-}
-
-function gatherPunctuatorsFromCurrentDeclaration(node, source) {
-  return node.declarations.reduce(function (tokens, subToken, index) {
-    // skip first one and empty declarations
-    if (subToken.init && index > 0) {
-      var punctuator = source.getTokensBetween(subToken.id, subToken.init)[0];
-      if (isPunctuatorAssignment(punctuator) && isMultipleLineDeclaration(node, punctuator)) {
-        tokens.push(punctuator);
+module.exports = function (context) {
+  var source = context.getSourceCode(), sourceLines = context.getSourceLines();
+
+  function declarationCheck(node) {
+    function findFirstDeclaredVar(node, steps) {
+      if (node.declarations.length > 1) {
+        var step = steps || 0;
+        if (typeof node.declarations[step] === 'undefined') {
+          return node.declarations[0];
+        }
+        if (node.declarations[step].init === null) {
+          return findFirstDeclaredVar(node, step + 1);
+        }
+        return node.declarations[step];
       }
+      return node.declarations[0];
     }
-    return tokens;
-  }, []);
-}
-
-function isNewVariable(expression) {
-  return expression.type === 'Keyword' && -1 !== ['var', 'let', 'const'].indexOf(expression.value);
-}
-
-function findNextPunctuators(node, source, lines) {
-  var after       = source.getTokenAfter(node),
-      nextLine    = lines[node.loc.start.line];
-
-  // in this case one big var assignment is declared:
-  //
-  // var var1 = {},
-  //     foo  = [];
-  // let blah = '';
-  var punctuators = node.declarations.length > 1 ? gatherPunctuatorsFromCurrentDeclaration(node, source) : [];
 
-  // check the next line for further expressions and skip completly in case of empty lines
-  if (typeof after !== 'undefined' && typeof nextLine !== 'undefined' && isNextLineEmpty(nextLine)) {
-    // checks whether a new variable is introduced
-    // or an existing one will be changed.
-    var next       = source.getTokenAfter(after),
-        punctuator = isNewVariable(after) ? source.getTokenAfter(next) : next;
-
-    if (isPunctuatorAssignment(punctuator)) {
-      punctuators.push(punctuator);
+    // skip if the variable is declared, but not assigned to a value
+    var first = findFirstDeclaredVar(node);
+    if (typeof first.init !== 'undefined' && first.init !== null) {
+      // parse through the next AST tokens in order
+      // to gather the next relevant tokens being necessary
+      // for the alignment validation.
+      var punctuator      = source.getTokensBetween(first.id, first.init).shift(),
+          nextPunctuators = findNextPunctuators(node, source, sourceLines);
+
+      if (typeof nextPunctuators !== 'undefined') {
+        validatePunctuators(punctuator, nextPunctuators, context, node);
```

Comment:
this part is partially duplicated with ``expressionCheck``, so it should be evaluated whether to abstract pieces of this code snippet